src/test/java/com/example/springbootbookshop/controller/BookControllerTest.java

```diff
@@ -0,0 +1,232 @@
+                .apply(springSecurity())
+                .build();
+        tearDown(dataSource);
+    }
+
+    @BeforeEach
+    void beforeEach(@Autowired DataSource dataSource) throws SQLException {
+        tearDown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/category/insert-category.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/insert-books.sql"));
+        }
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        tearDown(dataSource);
+    }
+
+    @SneakyThrows
+    static void tearDown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/category/clear-category.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/clear-books-db.sql"));
+        }
+    }
+
+    @Test
+    @DisplayName("Create a new book")
+    @Sql(scripts = "classpath:database/books/clear-books-db.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    void createNewBookAndSaveIntoDbAssertSuccess() throws Exception {
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto(
+                "Book1",
+                "Author1",
+                "ISBN1",
+                BigDecimal.TEN,
+                "Description1",
+                "coverImageUrl1",
+                List.of(1L));
+        BookDto expectedDto = getExpectDto(createBookRequestDto);
+        String jsonRequest = objectMapper.writeValueAsString(createBookRequestDto);
+
+        MvcResult result = mockMvc.perform(post("/books")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto actualDto = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+        assertNotNull(actualDto);
+        EqualsBuilder.reflectionEquals(expectedDto, actualDto,"id");
+    }
+
+    @Test
+    @DisplayName("Get all books")
+    @WithMockUser(username = "user")
+    void getAllBooksFromDbAssertSuccess() throws Exception {
+        List<BookDto> expectBookDtoList = getBookDtoList();
+
+        MvcResult result = mockMvc.perform(get("/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actualDtoList = objectMapper.readValue(result.getResponse().getContentAsString(),
```

Comment:
To get List<BookDto> is better to use 
```suggestion
        List<BookDto> actualDtoList = objectMapper.readValue(result.getResponse().getContentAsString(), 
        new TypeReference<List<BookDto>>() {});
```